### PR TITLE
feat(streamlit-demo): add Streamlit App

### PR DESCRIPTION
**Pull request by zebin-huang**
This PR introduces the initial Streamlit demo, including documentation, refactored TinyBERT model training, and fixes for deployment issues.

Based on #2, the issues with the API key and model deployment have been fixed, and the page code has been refactored.

### CODE REVIEW

**Comment by zebin-huang** [comment]
Hi, you can access this online demo here, https://github.com/zebin-huang/gsoc-streamlit/tree/streamlit/streamlit

**Review by sergiopaniego — COMMENTED** [review]
The branch includes some conflicts with the previous one. Could you resolve them or provide more details about them? 

image

**Comment by zebin-huang** [comment]
Hi @sergiopaniego  I've resolved all the conflicts.

File: streamlit/README.md [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+virtualenv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
+streamlit run streamlit/main.py
+```
 
-Deployment Notes
 - The model training process may be slow due to the limitations of Streamlit Community Cloud.
-- The model used for testing is large and cannot be hosted directly on the GitHub repository. Currently, this model is not included in the Streamlit deployment.
+- The model ([See huggingface models](https://huggingface.co/zebin-huang/gsoc2024-ZebinHuang/tree/main)) used for testing is large and be hosted directly on the GitHub repository.
```
Comment (sergiopaniego) [inline streamlit/README.md]: This sentence is unclear. We could rewrite it 😄 

`The model used for testing is large and is hosted directly on the GitHub repository.`

**Review by sergiopaniego — CHANGES_REQUESTED** [review]

File: streamlit/README.md (hunk 1 above)
Comment (zebin-huang) [reply streamlit/README.md]: Hi @sergiopaniego, I've updated the documentation. Please let me know if you encounter any other issues. 😄

**Review by sergiopaniego — CHANGES_REQUESTED** [review]
Instead of GitHub LFS, it is git lfs

References:
* https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage
* https://huggingface.co/docs/hub/repositories-getting-started

**Comment by sergiopaniego** [comment]
:rocket:

**Pull request merged**